mwdb/web/src/components/ShowObject/Views/SharesBox.js

```diff
@@ -0,0 +1,180 @@
+                    value={group}
+                    inputProps={{id: 'states-autocomplete'}}
+                    getItemValue={(item) => item}
+                    shouldItemRender={(item, value) => {
+                        return (item.toLowerCase().indexOf(value.toLowerCase()) !== -1);
+                    }}
+                    items={props.groups}
+                    onChange={event => setGroup(event.target.value)}
+                    onSelect={value => setGroup(value)}
+                    renderInput={(props) =>
+                        <div className="input-group">
+                            <input {...props} className='form-control' placeholder="Share with group"/>
+                            <div className="input-group-append">
+                                <input className="btn btn-outline-primary" type="submit" value="Add"/>
+                            </div>
+                        </div>
+                    }
+                    wrapperStyle={{display:"block"}}
+                    renderMenu={children =>
+                        <div className={"dropdown-menu " + (children.length !== 0 ? "show" : "")}>
+                            {
+                                children.map(c =>
+                                    <a key={c} href="#dropdown" className="dropdown-item" style={{"cursor": "pointer"}}>
+                                        {c}
+                                    </a>
+                                )
+                            }
+                        </div>
+                    }
+                    renderItem={(item, isHighlighted) => (
+                        <div
+                            className={`item ${isHighlighted ? 'item-highlighted' : ''}`}
+                            key={item}
+                        >{item}</div>
+                    )}
+            />
+        </form>
+    );
+}
+
+function SharesBox() {
+    const context = useContext(ObjectContext);
+
+    const [groups, setGroups] = useState([]);
+    const [items, setItems] = useState([]);
+    const [shareReceiver, setShareReceiver] = useState("");
+    const [isModalOpen, setIsModalOpen] = useState(false);
+
+    async function updateShares() {
+        try {
+            let response = await api.getObjectShares(context.object.id)
+            setGroups(response.data.groups);
+            setItems(response.data.shares);
+        } catch(error) {
+            context.setObjectError(error);
+        }
+    }
+
+    async function doShare(group) {
+        try {
+            setIsModalOpen(false);
+            await api.shareObjectWith(context.object.id, group)
+            updateShares();
+        } catch(error) {
+            context.setObjectError(error);
+        }
+    }
+
+    function handleShare(group) {
+        setIsModalOpen(true);
+        setShareReceiver(group);
+    }
+
+    const getShares = useCallback(updateShares, [])
+
+    useEffect(() => {
+        getShares();
+    }, [getShares])
```

Comment:
Shares should be updated as well when `context.object.id` has been changed.

1. Add a file
2. Add a child file
3. Go to the parent file
4. Click on child relationship in Related samples
5. Visible set of shares is wrong